### PR TITLE
[android] remove player from lockscreen if it is stopped. fix #47 

### LINKED ISSUE #47: [android] Lockscreen Wallpaper remains even after player stops.
Tested on Samsung S4/Android 5.0.1 with no SD Card without any screen lock apps/wallpaper apps.

When FrostWire is running, with the Player playing, paused, or stopped=exited, the phone's lock screen shows a track cover of last played artist. Once FrostWire is closed, the wallpaper disappears and reverts to system choice. 

FIX: It should not be showing the wallpaper on lock screen either at all or at least not when player is not actively playing. 

Several users reported as well. Cannot reproduce on Sony Xperia Z2/ Android 5.1.1. It has Sony Themes pre-installed and as default and I believe this is what's overwriting FrostWire in my case.

### PR DESCRIPTION
**Pull request by X-NicON**
fix issue #47 
Added update the lockscreen controls for stop action.
tested android 4.4.

### CODE REVIEW

**Comment by gubatron** [comment]
will test

**Comment by gubatron** [comment]
tmp_11136-screenshot_2016-02-02-09-48-00-1600925274
tmp_11136-screenshot_2016-02-02-09-48-22-1552145890

**Comment by gubatron** [comment]
the bug still remains.

When I long press on the player to "STOP" playback, and lock the screen, I still see the last's track's cover, as if it were paused.

I should only see the cover of a track if it's being played, or if the player is paused.

When we kill the playback completely, with an on stop (long press the pause button), we should clear the background. This patch does not address this issue, actually not sure what it does.

**Comment by X-NicON** [comment]
i hurried.
it was necessary to ask first.
if not difficult, even "pull request" will remain, a look later.

**Comment by gubatron** [comment]
good, please continue to work on this same branch, once it's all good, you give me a Bitcoin address and I'll send you the bounty.

**Comment by X-NicON** [comment]
now I figured out. Done and tested in different situations
Tested on Android 5.1.1

**Comment by gubatron** [comment]
just saw this, will test in about 2 hours.
Thank you.

File: android/apollo/src/com/andrew/apollo/MusicPlaybackService.java [hunk 1]
```diff
@@ -1357,7 +1365,7 @@ private void notifyChange(final String what) {
         // Update the lockscreen controls
         updateRemoteControlClient(what);
 
-        if (what.equals(POSITION_CHANGED)) {
+        if (what.equals(POSITION_CHANGED) || what.equals(PLAYSTATE_STOPPED)) {
```
Comment (gubatron) [inline android/apollo/src/com/andrew/apollo/MusicPlaybackService.java]: This code tells me that it was a mistake using `notifyChange(PLAYSTATE_STOPPED)`.

If all you actually needed was to call `updateRemoteControlClient(PLAYSTATE_STOPPED)`

File: android/apollo/src/com/andrew/apollo/MusicPlaybackService.java [hunk 2]
```diff
@@ -923,6 +930,7 @@ private void stop(final boolean goToIdle) {
         if (goToIdle) {
             scheduleDelayedShutdown();
             mIsSupposedToBePlaying = false;
+            notifyChange(PLAYSTATE_STOPPED);
```
Comment (gubatron) [inline android/apollo/src/com/andrew/apollo/MusicPlaybackService.java]: nope, instead just call `updateRemoteControlClient(PLAYSTATE_STOPPED)`

File: android/apollo/src/com/andrew/apollo/MusicPlaybackService.java [hunk 3]
```diff
@@ -801,6 +807,7 @@ private void handleCommandIntent(Intent intent) {
             mPausedByTransientLossOfFocus = false;
             seek(0);
             releaseServiceUiAndStop();
+            notifyChange(PLAYSTATE_STOPPED);
```
Comment (gubatron) [inline android/apollo/src/com/andrew/apollo/MusicPlaybackService.java]: instead just call `updateRemoteControlClient(PLAYSTATE_STOPPED)`

File: android/apollo/src/com/andrew/apollo/MusicPlaybackService.java [hunk 4]
```diff
@@ -604,6 +609,7 @@ private void initService() {
         reloadQueue();
         notifyChange(QUEUE_CHANGED);
         notifyChange(META_CHANGED);
+        notifyChange(PLAYSTATE_STOPPED);
```
Comment (gubatron) [inline android/apollo/src/com/andrew/apollo/MusicPlaybackService.java]: instead just call `updateRemoteControlClient(PLAYSTATE_STOPPED)`

File: android/apollo/src/com/andrew/apollo/MusicPlaybackService.java [hunk 5]
```diff
@@ -1410,15 +1418,18 @@ private void notifyChange(final String what) {
      * @param what The broadcast
      */
     private void updateRemoteControlClient(final String what) {
+
         int playState = mIsSupposedToBePlaying
-                ? RemoteControlClient.PLAYSTATE_PLAYING
+            ? RemoteControlClient.PLAYSTATE_PLAYING
+            : what.equals(PLAYSTATE_STOPPED)
+                ? RemoteControlClient.PLAYSTATE_STOPPED
                 : RemoteControlClient.PLAYSTATE_PAUSED;
```
Comment (gubatron) [inline android/apollo/src/com/andrew/apollo/MusicPlaybackService.java]: the block above should probably be rewritten with if statements for clarity.
Nested ternaries are a source of bugs and a pain to the eye.

I think, but I'm not sure, that we might be able to deduce that the player is stopped without having to add a new state.

If it's not playing, and it's not paused, isn't that the same as stopped?

Comment (X-NicON) [inline android/apollo/src/com/andrew/apollo/MusicPlaybackService.java]: Here need to accurately indicate the status. Therefore use a new status PLAYSTATE_STOPPED.

**Comment by X-NicON** [comment]
As a result, I understand, the problem in "killNotification()" that kill notification, while leaving cover. My commit, as if solves the problem, but he's not very good quality. Merge or not merge - you decide, I was just wondering.

**Comment by gubatron** [comment]
I've merged this @X-NicON.
Please send me a Bitcoin address to send your bounty today.

**Comment by gubatron** [comment]
Payment sent
https://blockchain.info/tx/5a8e8276c6fa424ad2792ce7063a3e98fb94659339f556f91cbcc6989bf0d3f4

**Pull request closed without merging**